weavelet.go

```diff
@@ -367,6 +390,11 @@ func (w *weavelet) getListener(name string, opts ListenerOptions) (*Listener, er
 	if reply.Error != "" {
 		return nil, fmt.Errorf("getListener(%q): %s", name, reply.Error)
 	}
+
+	ls := w.getListenerState(name)
+	ls.addr = l.Addr().String()
```

Comment:
Is it okay that the assignment here isn't guarded by a lock? Can concurrent calls to `getListener` with the same name race?

Reply:
Great catch. Fixed.

PS. I am not sure our listener management code works well in the face of multiple calls to get the listener. E.g., we might listen multiple times on the same address.

Reply:
Great point. Hopefully when we switch to the new listener API, we can also clean up some of these things.